quickwit/quickwit-storage/src/object_storage/s3_compatible_storage.rs

```diff
@@ -595,11 +599,26 @@ impl S3CompatibleObjectStorage {
                 unattempted.extend(chunk.iter().map(|path| path.to_path_buf()));
                 continue;
             }
-            let objects: Vec<ObjectIdentifier> = chunk
+            let Ok(objects): Result<Vec<ObjectIdentifier>, _> = chunk
                 .iter()
                 .map(|path| ObjectIdentifier::builder().key(self.key(path)).build())
-                .collect();
-            let delete = Delete::builder().set_objects(Some(objects)).build();
+                .collect()
+            else {
+                error = Some(
+                    StorageErrorKind::Internal
+                        .with_error(anyhow!("failed to build object identifier")),
+                );
+                unattempted.extend(chunk.iter().map(|path| path.to_path_buf()));
```

Comment:
that code logic is not good:
we batch things in chunk... If one of the object is wrong in the chunk we ditch the entire chunk, but not the entire request, even though at this point not request happened.

This does not make much sense. Chunking should happen after creating the object identifier. 
The code woudl be simpler and the spec would make way more sense.